src/main/kotlin/org/wa/rceditor/application/viewmodel/MainViewModel.kt

```diff
@@ -1,113 +1,320 @@
     }
 
     fun removeSource(item: SourceItem) {
-        model.removeSource(item)
+        sources.remove(item)
     }
 
-    fun checkingEntities(): ObservableList<CheckingEntityItem> {
-        return model.checkingEntities()
+    fun checkingEntities(): ObservableList<StringProperty> {
+        return checkingEntities
     }
 
     fun addCheckingEntity(text: String) {
-        model.addCheckingEntity(text)
+        checkingEntities.add(SimpleStringProperty(text))
     }
 
-    fun removeCheckingEntity(item: CheckingEntityItem) {
-        model.removeCheckingEntity(item)
+    fun removeCheckingEntity(item: StringProperty) {
+        checkingEntities.remove(item)
     }
 
     fun projects(): ObservableList<ProjectItem> {
-        return model.projects()
+        return projects
     }
 
     fun addProject(title: String, versification: String, identifier: String, sort: Int, path: String, category: String) {
-        model.addProject(title, versification, identifier, sort, path, category)
+        projects.add(ProjectItem(title, versification, identifier, sort, path, category))
     }
 
     fun removeProject(item: ProjectItem) {
-        model.removeProject(item)
+        projects.remove(item)
     }
 
-    // ------------- Handlers ---------------- //
+    fun loadRecourceContainer() {
+        conformsto = container.manifest.dublinCore.conformsTo
+        creator = container.manifest.dublinCore.creator
+        description = container.manifest.dublinCore.description
+        format = container.manifest.dublinCore.format
+        identifier = container.manifest.dublinCore.identifier
+        issued = LocalDate.parse(container.manifest.dublinCore.issued, DateTimeFormatter.ISO_DATE)
+        modified = LocalDate.parse(container.manifest.dublinCore.modified, DateTimeFormatter.ISO_DATE)
+        languageDirection = container.manifest.dublinCore.language.direction
+        languageIdentifier = container.manifest.dublinCore.language.identifier
+        languageTitle = container.manifest.dublinCore.language.title
+        publisher = container.manifest.dublinCore.publisher
+        rights = container.manifest.dublinCore.rights
+        subject = container.manifest.dublinCore.subject
+        title = container.manifest.dublinCore.title
+        type = container.manifest.dublinCore.type
+        version = container.manifest.dublinCore.version
+        checkingLevel = container.manifest.checking.checkingLevel
 
-    fun handleNewDocumentSelected() {
-        model.onNewDocumentSelected()
+        contributors.addAll(container.manifest.dublinCore.contributor.map { SimpleStringProperty(it) })
+
+        container.manifest.dublinCore.relation.forEach {
+            addRelation(it)
+        }
+
+        container.manifest.dublinCore.source.forEach {
+            addSource(it.identifier, it.language, it.version)
+        }
+
+        container.manifest.checking.checkingEntity.forEach {
+            addCheckingEntity(it)
+        }
```

Comment:
If you haven't already, you can modify lines 206-216 to be more like 204.

Reply:
@mbr4477 Done!